### PR TITLE
[css-ruby] Added 1 intra-base test

### PR DESCRIPTION
**Pull request by TalbotG**
This is a followup to #29844

At my website:

http://www.gtalbot.org/BrowserBugsSection/CSS3Ruby/intra-base-white-space-001.html

http://www.gtalbot.org/BrowserBugsSection/CSS3Ruby/reference/intra-base-white-space-001-ref.html

A single 'white-space: normal' white space within a ruby base unit must not be discarded or removed; it must act in pairing base with text annotation.

### CODE REVIEW

**Review by upsuper — COMMENTED** [review]
I don't understand what this is trying to check. What could potentially go wrong here? Treating `<rb>` as a block container and thus trim the whitespaces?

There doesn't seem to be any spec text specifically mention this, and as far as `<rb>`s are treated like inline element, I don't quite see how this could go wrong :thinking:

**Comment by fantasai** [comment]
@upsuper Does seem like that's what it's checking. It'd be maybe more interesting if it were written as `  <ruby>b<rb> </rb> <rb> b</rb><rt>a</rt><rt></rt><rt>a</rt></ruby>` though, since that also checks that things collapse correctly. :)

**Comment by TalbotG** [comment]
> ... more interesting if it were written as ` <ruby>b<rb> </rb> <rb> b</rb><rt>a</rt><rt></rt><rt>a</rt></ruby>` though, since that also checks that things collapse correctly. :)

Yes, I can do that later next month, in october.

**Comment by TalbotG** [comment]
At my website, commit [96c7c28](https://github.com/web-platform-tests/wpt/pull/30238/commits/96c7c289487a35027dc95e18a002f339ca645ac0) correspondent files are:

http://www.gtalbot.org/BrowserBugsSection/CSS3Ruby/intra-base-white-space-001v2.html

http://www.gtalbot.org/BrowserBugsSection/CSS3Ruby/reference/intra-base-white-space-001v2-ref.html

I have made ruby base units more easily differentiable, distinguishible among themselves by using different characters (b, c, d and e). 

I also added the code scenario where there is a trailing white space inside a ruby base unit with such white space not being the last character of a ruby base container. This is the `<rb>d </rb>` one.

The `<rb> c</rb>` one is the leading code scenario.

I have spent quite some time reading the ruby spec and I am _not_ perfectly sure about this test. So, do not hesitate if you believe - **and can explain why** - the white space handling of such test is incorrect.

**Comment by fantasai** [comment]
@TalbotG Your diagram misses the empty base-annotation pair right before the last a/c pair, but otherwise I think it's correct.

**Review by fantasai — APPROVED** [review]
Please fix the diagram before merging. :)

File: css/css-ruby/intra-base-white-space-001.html [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    b<rb> </rb><rb> c</rb><rb>d </rb><rb> </rb><rb>e</rb>
+
+    <rt>a</rt><rt></rt><rt>a</rt><rt>a</rt><rt></rt><rt>a</rt>
+
+  </ruby>
+
+  <!--
+
+  |[ a ]|[  ]|[ a ]|  [ a ]  |[ a ]|
+  |[ b ]|[ws]|[ c ]|[ d ][ws]|[ e ]|
```
Comment (fantasai) [inline css/css-ruby/intra-base-white-space-001.html:33]: There's an empty base + empty annotation pair right before the last one, so the diagram should be this:

```suggestion
  |[ a ]|[  ]|[ a ]|  [ a ]  |[]|[ a ]|
  |[ b ]|[ws]|[ c ]|[ d ][ws]|[]|[ e ]|
```

**Pull request merged**